### PR TITLE
Fix of time zone issues

### PR DESCRIPTION
**Pull request by EinKrebs**

### CODE REVIEW

File: storage/clickhousespanstore/reader.go [hunk 1]
```diff
@@ -309,10 +309,10 @@ func (r *TraceReader) findTraceIDsInRange(ctx context.Context, params *spanstore
 	}
 
 	query += " AND -toUnixTimestamp(timestamp) <= -toUnixTimestamp(?)"
-	args = append(args, start.UTC().Format("2006-01-02T15:04:05"))
```
Comment (pavolloffay) [inline storage/clickhousespanstore/reader.go:312]: Could you explain why the format change is needed?

File: cmd/jaeger-clickhouse/main.go [hunk 2]
```diff
@@ -6,6 +6,9 @@ import (
 	"os"
 	"path/filepath"
 
+	// Package contains time zone info for connecting to ClickHouse servers with non-UTC time zone
+	_ "time/tzdata"
```
Comment (pavolloffay) [inline cmd/jaeger-clickhouse/main.go:10]: > // Package tzdata provides an embedded copy of the timezone database.
// If this package is imported anywhere in the program, then if
// the time package cannot find tzdata files on the system,
// it will use this embedded information.

File: storage/clickhousespanstore/reader.go (hunk 1 above)
Comment (EinKrebs) [reply storage/clickhousespanstore/reader.go:312]: It's needed because search on clickhouse severs with non-UTC timezones works incorrectly. It happens because formatting  time the previous way deleted information about time zone, so search range "shifted" by differ between UTC and server time zone. Furthermore, passing time.Time as query parameter works perfectly fine.

File: cmd/jaeger-clickhouse/main.go (hunk 2 above)
Comment (EinKrebs) [reply cmd/jaeger-clickhouse/main.go:10]: To connect to ClickHouse server with non-UTC time zone, you need to have information about this time zone. Importing this library is universal way to provide such information on every system.

File: storage/clickhousespanstore/reader.go (hunk 1 above)
Comment (pavolloffay) [reply storage/clickhousespanstore/reader.go:312]: So line 312 adds the argument for the query on line 311 - `toUnixTimestamp(?)`. format `"2006-01-02T15:04:05"` does not encode timezone. Could we use a format that adds timezone? Then this is query is executed on clickhouse server.

I would prefer to be explicit and pass timestamp string instead of time instance.

Comment (pavolloffay) [reply storage/clickhousespanstore/reader.go:312]: maybe we will also have to make sure the timestamp format on the writer side matches this format

Comment (EinKrebs) [reply storage/clickhousespanstore/reader.go:312]: Writer side doesn't format time too, and I think that it's fine to relay on clickhouse-go driver on this.

Comment (EinKrebs) [reply storage/clickhousespanstore/reader.go:312]: Clickhouse driver for database/sql supports passing time.Time as argument to statement.Exec(), and I passing it by formatting is a bit of wheel inventing.

Comment (pavolloffay) [reply storage/clickhousespanstore/reader.go:312]: Alright, let's take this approach.

File: config.yaml [hunk 3]
```diff
@@ -7,3 +7,13 @@ batch_write_size:
 batch_flush_interval:
 # Encoding of stored data. Either json or protobuf. Default json.
 encoding:
+# Whether to use TLS connection. Default false.
+tls_connection:
```
Comment (pavolloffay) [inline config.yaml:11]: why are these changes here again? Didn't we merge this before?

Comment (EinKrebs) [reply config.yaml:11]: We definitely did, I don't know it's marked as a change

Comment (pavolloffay) [reply config.yaml:11]: By looking at the commits the PR includes some that should not be here https://github.com/pavolloffay/jaeger-clickhouse/pull/26/commits. E.g. The "expose metrics"

Comment (pavolloffay) [reply config.yaml:11]: nit: when working on a PR create a branch from the main/head`

```
git pull upstream main
git checkout -b feature-name
```

Then if there is a conflict or you want to synch with the main branch use rebase:
```
git rebase -i upstream main
```

Before running the rebase I would recommend squashing all commits (git rebase -i <hash of the commit>)

Comment (EinKrebs) [reply config.yaml:11]: > By looking at the commits the PR includes some that should not be here https://github.com/pavolloffay/jaeger-clickhouse/pull/26/commits. E.g. The "expose metrics"

It actually happen due to rebasing branch. Maybe I did it incorrectly, gonna try redoing it properly

Comment (EinKrebs) [reply config.yaml]: > gonna try redoing it properly

Done

**Review by pavolloffay — APPROVED** [review]
